File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeTwoLists(ListNode list1, ListNode list2) {
        ListNode ans=null, last=null;
        while (list1!=null || list2!=null) {
            ListNode oldlast=last;
            if (list1==null) {
                last=new ListNode(list2.val, null);
                list2=list2.next;
            }
            else if (list2==null) {
                last=new ListNode(list1.val, null);
                list1=list1.next;
            }
            else if (list1.val>list2.val) {
                last=new ListNode(list2.val, null);
                list2=list2.next;
            }
            else {
                last=new ListNode(list1.val, null);
                list1=list1.next;
            }
            if (ans==null) ans=last;
            else oldlast.next=last;
        }

        return ans;
    }
}

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        # ans=ListNode(0, None)
        # tmp=ans
        # while list1!=None and list2!=None:
        #     if list1.val>list2.val:
        #         tmp.next=ListNode(list2.val, None)
        #         tmp=tmp.next
        #         list2=list2.next
        #     else:
        #         tmp.next=ListNode(list1.val, None)
        #         tmp=tmp.next
        #         list1=list1.next
        # if list1!=None:
        #     tmp.next=list1
        # elif list2!=None:
        #     tmp.next=list2
        # return ans.next
        if list1 is None:
            return list2
        if list2 is None:
            return list1
        
        if list1.val>list2.val:
            list2.next=self.mergeTwoLists(list1, list2.next)
            return list2
        else:
            list1.next=self.mergeTwoLists(list1.next, list2)
            return list1